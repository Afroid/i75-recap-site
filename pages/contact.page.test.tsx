import React from "react";
import { render, screen } from "@testing-library/react";
import ContactPage from "@/pages/contact.page";

// Stub out next/head so it just renders its children
jest.mock("next/head", () => {
  return {
    __esModule: true,
    default: ({ children }: { children: React.ReactNode }) => <>{children}</>,
  };
});

describe("ContactPage", () => {
  it("renders the mailto link correctly", () => {
    render(<ContactPage />);
    const emailLink = screen.getByTestId("contact-email-link");
    const form = screen.getByTestId("contact-form");
    const formNameSection = screen.getByTestId("contact-form-name-section");
    const formEmailSection = screen.getByTestId("contact-form-email-section");
    const formMessageSection = screen.getByTestId("contact-form-message-section");
    const formSubmit = screen.getByTestId("contact-form-submit");

    // Assertions for email
    expect(emailLink).toBeInTheDocument();
    expect(emailLink).toHaveAttribute("href", "mailto:notCurrentlyActiveEmail@i75league.com");
    expect(emailLink).toHaveTextContent("notCurrentlyActiveEmail@i75league.com");

    // Assertions for form, sections, and the submit button
    expect(form).toBeInTheDocument();
    expect(formNameSection).toBeInTheDocument();
    expect(formEmailSection).toBeInTheDocument();
    expect(formMessageSection).toBeInTheDocument();
    expect(formSubmit).toBeInTheDocument();
  });

  it("matches the snapshot", () => {
    const { asFragment } = render(<ContactPage />);
    expect(asFragment()).toMatchSnapshot();
  });
});
