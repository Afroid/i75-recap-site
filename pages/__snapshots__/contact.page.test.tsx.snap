// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactPage matches the snapshot 1`] = `
<DocumentFragment>
  <title>
    Contact â€“ I75 League Recaps
  </title>
  <meta
    content="Get in touch with the I75 League recap team."
    name="description"
  />
  <main
    class="pt-6 max-w-2xl mx-auto px-4"
    data-testid="contact-main"
  >
    <h1
      class="text-4xl font-bold mb-4"
    >
      Contact Us
    </h1>
    <p
      class="text-lg mb-6"
    >
      For questions about the site, to join our league, or to share feedback, drop us a line:
    </p>
    <p
      class="mb-8"
    >
      <a
        class="text-green-600 hover:underline"
        data-testid="contact-email-link"
        href="mailto:notCurrentlyActiveEmail@i75league.com"
      >
        notCurrentlyActiveEmail@i75league.com
      </a>
    </p>
    <form
      class="space-y-4"
      data-testid="contact-form"
    >
      <div
        data-testid="contact-form-name-section"
      >
        <label
          class="block mb-1 font-medium"
          for="name"
        >
          Name
        </label>
        <input
          class="w-full border px-3 py-2 rounded-md"
          id="name"
          required=""
          type="text"
        />
      </div>
      <div
        data-testid="contact-form-email-section"
      >
        <label
          class="block mb-1 font-medium"
          for="email"
        >
          Email
        </label>
        <input
          class="w-full border px-3 py-2 rounded-md"
          id="email"
          required=""
          type="email"
        />
      </div>
      <div
        data-testid="contact-form-message-section"
      >
        <label
          class="block mb-1 font-medium"
          for="message"
        >
          Message
        </label>
        <textarea
          class="w-full border px-3 py-2 rounded-md"
          id="message"
          required=""
          rows="4"
        />
      </div>
      <button
        class="px-6 py-3 bg-green-600 text-white rounded-md hover:bg-green-700 transition"
        data-testid="contact-form-submit"
        type="submit"
      >
        Send Message - Not Functional
      </button>
    </form>
  </main>
</DocumentFragment>
`;
